src/wallet/wallet.cpp

```diff
@@ -3027,10 +3032,8 @@ bool CWallet::CreateTransactionInternal(
                             break;
                         }
 
-                        // The change value was not enough to cover the fees, even after dropping the change
-                        // We set nFeeRet back to its original value (the one that includes the change output) and try selecting again
+                        // The change value was not enough to cover the fees, even after dropping the change. Try selecting again.
                         assert(change_and_fee < nFeeRet);
-                        nFeeRet = orig_fee;
```

Comment:
In commit "Roll static tx fees into nValueToSelect instead of having it be separate" (67fc8b7be1d5fbdb199d6e995434431a7282d446)

I could be thinking about this wrong, but this change seems backwards. It matches the description which says "use the changeless nFeeRet when iterating for KnapsackSolver. This is because we include
the change fee when doing KnapsackSolver, so nFeeRet on furtheriterations won't include the change fee."

But isn't this the opposite of what we want? If KnapSackSolver is going to find a solution requiring a change output, shouldn't nFeeRet be set to the full cost of a transaction including the change output (orig_fee)? Not the cost of the smaller transaction without a change output.

Reply:
It's because we are including the change fee during `SelectCoinsMinConf`. So using the fee that includes the change fee results in us targeting a slightly higher value than we actually need. This is also moot because we stop the looping behavior in the next two commits that have KnapsackSolver use effective values.